public/components/event_analytics/utils/utils.tsx

```diff
@@ -442,3 +442,15 @@ export const redoQuery = (
     setData(res.jsonData);
   });
 };
+
+// method to return the type of a field from its name
+export const getFieldTypes = (newFieldName: string, explorerFields: IExplorerFields) => {
+  let fieldType: string = '';
```

Comment:
we don't need this line, we can actually turn it into something 

```
export const getFieldTypes = (newFieldName: string, explorerFields: IExplorerFields): string => {
  for (const field of explorerFields.availableFields) {
    if (field.name === newFieldName) return field.type;
  }
  for (const field of explorerFields.selectedFields) {
    if (field.name === newFieldName) return field.type;
  }
  return '';
}
```

Reply:
modified, and now using forEach instead of map